server/src/main/java/org/opensearch/common/cache/store/enums/CacheStoreType.java

```diff
@@ -0,0 +1,20 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.common.cache.store.enums;
+
+/**
+ * Cache store types in tiered cache.
+ *
+ * @opensearch.internal
+ */
+public enum CacheStoreType {
+
+    ON_HEAP,
+    DISK;
```

Comment:
You might want to consider an off-heap cache as an intermediate cache store for faster access which pays the ser/de penalty.

Reply:
Thanks for the suggestion @Bukhtawar! Had looked at that paper earlier, interesting read.

But as part of phase-1, we are only considering disk as a cache type. Will consider off heap in next phases.

Reply:
I captured it [here](https://github.com/opensearch-project/OpenSearch/issues/10024) in the main issue link under Milestone 3.